consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinatorTest.java

```diff
@@ -200,6 +222,85 @@ public void coinbaseShouldMatchSuggestedFeeRecipient() {
+
+    MergeCoordinator.MergeBlockCreatorFactory mergeBlockCreatorFactory =
+        (parentHeader, address) -> {
+          MergeBlockCreator beingSpiedOn =
+              spy(
+                  new MergeBlockCreator(
+                      address.or(miningParameters::getCoinbase).orElse(Address.ZERO),
+                      () -> Optional.of(30000000L),
+                      parent -> Bytes.EMPTY,
+                      transactions,
+                      protocolContext,
+                      protocolSchedule,
+                      this.miningParameters.getMinTransactionGasPrice(),
+                      address.or(miningParameters::getCoinbase).orElse(Address.ZERO),
+                      this.miningParameters.getMinBlockOccupancyRatio(),
+                      parentHeader));
+
+          doCallRealMethod()
+              .doCallRealMethod()
+              .doThrow(new MerkleTrieException("missing leaf"))
+              .doCallRealMethod()
+              .when(beingSpiedOn)
+              .createBlock(any(), any(Bytes32.class), anyLong());
+          return beingSpiedOn;
+        };
+
+    MergeCoordinator willThrow =
+        spy(
+            new MergeCoordinator(
+                protocolContext,
+                protocolSchedule,
+                proposalBuilderExecutor,
+                miningParameters,
+                backwardSyncContext,
+                mergeBlockCreatorFactory));
+
+    final AtomicLong retries = new AtomicLong(0);
+    doAnswer(
+            invocation -> {
+              if (retries.getAndIncrement() < txPerBlock) {
+                // a new transaction every time a block is built
+                transactions.addLocalTransaction(
+                    createTransaction(retries.get() - 1), Optional.empty());
+              } else {
+                // when we have 5 transactions finalize block creation
+                willThrow.finalizeProposalById(invocation.getArgument(0, PayloadIdentifier.class));
+              }
+              return null;
+            })
+        .when(mergeContext)
+        .putPayloadById(any(), any());
+
+    var payloadId =
+        willThrow.preparePayload(
+            genesisState.getBlock().getHeader(),
+            System.currentTimeMillis() / 1000,
+            Bytes32.random(),
+            suggestedFeeRecipient);
+
+    blockCreationTask.get();
+
+    ArgumentCaptor<Block> block = ArgumentCaptor.forClass(Block.class);
+
+    verify(willThrow, never()).addBadBlock(any(), any());
+    verify(mergeContext, times(txPerBlock + 1))
+        .putPayloadById(eq(payloadId), block.capture()); // +1 for the empty
+    assertThat(block.getValue().getBody().getTransactions().size()).isEqualTo(txPerBlock);
+    // this only verifies that adding the bad block didn't happen through the mergeCoordinator, it
+    // still may
+    // be called directly.
+    verify(badBlockManager, never()).addBadBlock(any(), any());
+    verify(willThrow, never()).addBadBlock(any(), any());
```

Comment:
this seems a duplication of line 293

Reply:
good catch! the intent was to make sure nothing was added to the badblock manager both before and after processing.  I've fixed this.